### PR TITLE
Allow the consumer to inject widgets into each row of an upload file list.

### PR DESCRIPTION
**Pull request by subdavis**
Work in progress.

Block on #154

### CODE REVIEW

File: src/components/Upload.vue [hunk 1]
```diff
@@ -28,8 +28,8 @@ v-card.fill-height(flat)
       v-alert(:value="true", dark, type="error") {{ errorMessage }}
         v-btn(v-if="!uploading", dark, outlined, @click="start") Resume upload
 
-    slot(name="files")
-      file-upload-list(v-model="files", v-bind="{ currentIndex, maxShow }")
+    slot(name="files", v-bind="{ files, setFiles, currentIndex, maxShow }")
+      file-upload-list(@input="setFiles", v-bind="{ value: files, currentIndex, maxShow }")
```
Comment (subdavis) [inline src/components/Upload.vue]: This is actually a separate bugfix.  If files are deleted such that index >= files.length, the index should be changed.

**Comment by zachmullen** [comment]
Do you have a screenshot for posterity?

**Comment by subdavis** [comment]
Examaple usage: 

Screenshot from 2019-09-27 11-19-45

**Comment by zachmullen** [comment]
Thanks! It's great to see this library proving its mission in the wild :)

**Pull request merged**